railties/lib/rails/generators.rb

```diff
@@ -281,6 +287,11 @@ def invoke(namespace, args = ARGV, config = {})
         end
       end
 
+      def add_generated_file(file) # :nodoc:
+        (@@generated_files ||= []) << file
```

Comment:
It feels a little strange to have this as global state.  I'm not sure how it will play with the [`:inline` option](https://github.com/rails/rails/blob/b69ac962adeef897d4d1ba9aac40b9501ce25659/railties/lib/rails/generators/actions.rb#L252-L259) added in #37516.  If there is no way for this to be instance state on the instantiated generator, what do you think about clearing `@@generated_files` after all callbacks are run?

Reply:
Thanks, your advice is correct and clearing `@@generated_files` was needed.  I fixed.